### PR TITLE
feat(argo-cd): Support ability to set .Values.namespaceOverride

### PR DESCRIPTION
**Pull request by andres-vara**
Resolves [#2664](https://github.com/argoproj/argo-helm/issues/2664)

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

**Comment by mkilchhofer** [comment]
Hi @andres-vara 

Since your PR description misses context or a related issue link:
What is your motivation resp. your use case for this change?

**Comment by andres-vara** [comment]
> Hi @andres-vara
> 
> Since your PR description misses context or a related issue link: What is your motivation resp. your use case for this change?

Hi @mkilchhofer
Attached [issue](https://github.com/argoproj/argo-helm/issues/2664) to the context.

In my case, I don't have access to .Release.Namespace via the CD tool we use, which is why I believe it's beneficial to have an alternative method of specifying the namespace using the values file.

**Comment by pdrastil** [comment]
@andres-vara Hi thanks for the contribution but Helm natively supports `helm install --namespace <target>` flag so IMHO I don't see a need to have override via values.yaml. Why is this needed?

**Comment by tico24** [comment]
@pdrastil I can answer this one.

If you use a chart as a dependent chart in another, you can't use the --namespace flag to put the dependent chart in a namespace different from the 'main' chart. This is extremely frustrating.

**Comment by pdrastil** [comment]
@tico24 Question is why would you want to do that. You don't deploy server / repo-server / each controller into separate namespace. In my experienece if you need more than 1 namespace then you should have 2 independent deployments instead of 1 huge umbrella chart as each namespace provides isolation (and if you have Cilium / NetPolicies the namespace override will not work reasonably well anyway).

**Comment by tico24** [comment]
I wouldn't want to do that. But I would want to deploy (in my case) argo workflows and another application in the same helm chart, but in different namespaces.

I can easily see a requirement to deploy argocd and another application in one helm chart but in different namespaces.

**Comment by andres-vara** [comment]
> @andres-vara Hi thanks for the contribution but Helm natively supports `helm install --namespace <target>` flag so IMHO I don't see a need to have override via values.yaml. Why is this needed?

As mentioned earlier, if your company policy requires all helm releases to be in one default namespace, '.Release.Namespace' may not be available for modification. There is no harm in having an additional option to manage the deployment namespace via values

**Comment by pdrastil** [comment]
Ok thanks for sharing context and I'm fine with having extra way of managing things.

File: charts/argo-cd/templates/argocd-application-controller/deployment.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ metadata:
     {{- end }}
   {{- end }}
   name: {{ template "argo-cd.controller.fullname" . }}
-  namespace: {{ .Release.Namespace | quote }}
+  namespace: {{ include "argo-cd.namespace" . | quote }}
```
Comment (pdrastil) [inline charts/argo-cd/templates/argocd-application-controller/deployment.yaml:12]: One small nit. Do we need `quote` here? We don't have it anyway in RBAC bindings.

**Comment by yu-croco** [comment]
Hi @andres-vara 
> I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).

can you please update (remove old ones and add new one) changelog in Chart.yaml ?
Ref: https://github.com/argoproj/argo-helm/pull/2679/files#diff-16f38cd1a4674cb682ac9f015fbc1c1ff552f024a8f791c16de0de21a1f65771R28-R34

**Review by pdrastil — APPROVED** [review]

**Pull request merged**